cmd/etcd-launcher/main.go

```diff
@@ -93,9 +101,30 @@ func main() {
 		log.Panicw("failed to set initialState", zap.Error(err))
 	}
 
+	// "strict" mode (only TLS for peer traffic) will be used for new clusters
+	e.usePeerTLSOnly = e.initialState == initialStateNew
+
+	// in addition, if "strict" mode is enforced, set it up for existing clusters too
+	if os.Getenv(envPeerTLSMode) == peerTLSModeStrict {
+		e.usePeerTLSOnly = true
+	}
+
 	log.Info("initializing etcd..")
 	log.Infof("initial-state: %s", e.initialState)
-	log.Infof("initial-cluster: %s", strings.Join(initialMemberList(e.clusterSize, e.namespace), ","))
+	log.Infof("initial-cluster: %s", strings.Join(initialMemberList(e.clusterSize, e.namespace, e.usePeerTLSOnly), ","))
+	if e.usePeerTLSOnly {
+		log.Info("peer-tls-mode: strict")
+	}
+
+	// if the cluster already exists, try to connect and update peer URLs that might be out of sync.
+	// etcd might fail to start if peer URLs in the etcd member state and the flags passed to it are different
```

Comment:
Would it be possible to check if all the members are available before we do this? Because if not, they won't be all updated and it might break the ring.

Reply:
Since this happens before the etcd node this `etcd-launcher` instance manages even starts, I'm not sure that's possible, there's going to be one unavailable by logic - or do you mean check the remaining members (so, 2 out of 3, 4 out of 5, etc) are available?

Reply:
No checking the remaining ones wouldn't make sense. They also might not be available.

Reply:
I moved it to before the start command because etcd can fail to start if the peer URLs in the cluster state of the node is different from what is passed via flags/environment variables. So the idea was to make sure the in-cluster state is in sync with the intention of the starting etcd node. Maybe run the peer URL update a second time after starting?